lib/desktop/session.rb

```diff
@@ -143,6 +144,10 @@ def clean
       FileUtils.rm_rf(session_dir_path)
     end
 
+    def geometry
+      local? ? @geometry : 'UNKNOWN'
```

Comment:
The reason that I said that remote geometries would be unknown is that I had assumed that you'd use `xrandr` to discover the current geometry.  Using `xrandr` handles the following use case:

1. User starts a desktop session
2. User connects to it and uses `xrandr` to change its geometry.
3. User connects to it via the webapp
4. The webapp queries the CLI for the geometry; the CLI reports its stale geometry.
5. The webapp informs the user of a stale geometry.

Did you consider using `xrandr`?

The current geometry is on the first line of output of `xrandr`.  The script `~/.local/share/flight/desktop/bin/flight-desktop_geometry.sh` has a (possibly helpful) example of how to extract it.